allennlp/modules/conditional_random_field_wemission.py

```diff
@@ -0,0 +1,82 @@
+"""
+Conditional random field
+"""
+from typing import List, Tuple
+
+import torch
+
+from allennlp.common.checks import ConfigurationError
+
+from .conditional_random_field import ConditionalRandomField
+
+
+class ConditionalRandomFieldWeightEmission(ConditionalRandomField):
+    """
+    This module uses the "forward-backward" algorithm to compute
+    the log-likelihood of its inputs assuming a conditional random field model.
+
+    See, e.g. http://www.cs.columbia.edu/~mcollins/fb.pdf
```

Comment:
Could you add something about the weighting strategy here?

Reply:
Done!